catalog/ui/catalog-ui-search/src/main/webapp/js/filter.js

```diff
@@ -0,0 +1,372 @@
+                    break;
+                case '<':
+                    if (matchesLessThan(valuesToCheck[i], filter)) {
+                        return true;
+                    }
+                    break;
+                case '<=':
+                    if (matchesLessThanOrEqualTo(valuesToCheck[i], filter)) {
+                        return true;
+                    }
+                    break;
+                case 'INTERSECTS':
+                    if (matchesPOLYGON(valuesToCheck[i], filter)) {
+                        return true;
+                    }
+                    break;
+                case 'DWITHIN':
+                    if (CQLUtils.isPointRadiusFilter(filter)) {
+                        if (matchesCIRCLE(valuesToCheck[i], filter)) {
+                            return true;
+                        }
+                    } else if (matchesLINESTRING(valuesToCheck[i], filter)) {
+                        return true;
+                    }
+                    break;
+                case 'AFTER':
+                    if (matchesAFTER(valuesToCheck[i], filter)) {
+                        return true;
+                    }
+                    break;
+                case 'BEFORE':
+                    if (matchesBEFORE(valuesToCheck[i], filter)) {
+                        return true;
+                    }
+                    break;
+            }
+        }
+        return false;
+    } else {
+        return matchesFilters(metacard, filter);
+    }
+}
+
+function matchesFilters(metacard, resultFilter) {
+    var i;
+    switch (resultFilter.type) {
+        case 'AND':
+            for (i = 0; i <= resultFilter.filters.length - 1; i++) {
+                if (!matchesFilter(metacard, resultFilter.filters[i])) {
+                    return false;
+                }
+            }
+            return true;
+        case 'NOT AND':
+            for (i = 0; i <= resultFilter.filters.length - 1; i++) {
+                if (!matchesFilter(metacard, resultFilter.filters[i])) {
+                    return true;
+                }
+            }
+            return false;
+        case 'OR':
+            for (i = 0; i <= resultFilter.filters.length - 1; i++) {
+                if (matchesFilter(metacard, resultFilter.filters[i])) {
+                    return true;
+                }
+            }
+            return false;
+        case 'NOT OR':
+            for (i = 0; i <= resultFilter.filters.length - 1; i++) {
+                if (matchesFilter(metacard, resultFilter.filters[i])) {
+                    return false;
+                }
+            }
+            return true;
+        default:
+            return matchesFilter(metacard, resultFilter);
+    }
+}
+
+module.exports = {
+    matchesFilters: matchesFilters,
+    matchesCql: function(metacardJSON, cqlString) {
+        if (cqlString === "") {
+            return true;
+        }
+        return this.matchesFilters(metacardJSON, cql.read(cqlString));
```

Comment:
The body of this function can be simplified to `return cqlString === "" || this.matchesFilters(metacardJSON, cql.read(cqlString));`.

Reply:
While entirely possible, I think it makes the code less readable.